include/exadg/incompressible_navier_stokes/spatial_discretization/operator_coupled.h

```diff
@@ -68,17 +73,24 @@ class NonlinearOperatorCoupled
    * 'evaluate_residual'.
    */
   void
-  evaluate_residual(BlockVectorType & dst, BlockVectorType const & src) const
+  evaluate_residual(BlockVectorType & dst, BlockVectorType const & src)
   {
+    // update implicitly coupled viscosity
+    if(nonlinear_viscous_term_treated_implicitly == true)
+    {
+      pde_operator->update_viscosity(src.block(0));
+    }
```

Comment:
This should happen inside `evaluate_nonlinear_residual()`. `NonlinearOperatorCoupled` is a pure "linear algebra" wrapper without any physics implemented.